### DISCUSSION TITLE
boot nodes get rewards, but staked validators don't seem to

### DISCUSSION THREAD

**Discussion by ghost** [0xPolygon/polygon-edge#945] [General]
# [ Subject of the issue ]
## Description
Describe your issue in as much detail as possible here. 2.5 hours ago we set up 10 bootnodes and 10 staked validators (using the 0x00000000000000000001001 address). The bootnodes get more coins every few minutes, but so far the staked validators have received none. All stakers and bootnodes have 5 million coins in each staked wallet, so it should be roughly equal. Am I missing a switch somewhere?

## Your environment
- OS and version. ubuntu 20.04
- The version of the Polygon Edge.    
  (*Confirm the version of your Polygon edge client by running the following command: `polygon-edge version --grpc-address GRPC_ADDRESS`*)
- The branch that causes this issue.
- Locally or Cloud hosted (which provider). cloud AWS
- Please confirm if the validators are running under containerized environment (K8s, Docker, etc.). tmux

## Steps to reproduce
- Tell us how to reproduce this issue. stake 5 million on a boot node, 5 million to the staking contract wallet from a validator and trow a few transactions across the network.
- Where the issue is, if you know.
- Which commands triggered the issue, if any.
- Provide us with the content of your genesis file.
- Provide us with commands that you used to start your validators.
- Provide us with the peer list of each of your validators by running the following command: `polygon-edge peers list --grpc-address GRPC_ADDRESS`.
- Is the chain producing blocks and serving customers atm?

## Expected behavior
- Tell us what should happen. everyone gets rewards, but the boots get slightly more for handling that particular block
- Tell us what happened instead.

## Logs
Provide us with debug logs from all of your validators by setting logging to `debug` output with: `server --log-level debug`

## Proposed solution
If you have an idea on how to fix this issue, please write it down here, so we can begin discussing it

**Comment by laviniat1996** [comment]
@Nexsuperne101 Did you generate keys for 10 nodes in the begining? Does your network have only 10 nodes? I suppose you don't have 20 nodes, right?

**Comment by ghost** [comment]
I generated keys for 10 nodes only, which are the 10 boot nodes. Then I flipped from PoA to PoS, put 5 million coins into each boot node, went to PoS successfully. Now we are staking 5 million coins from each staker wallet to the 0x0000000000000000000000000000000000001001
address via a DApp. Is this not right? Does each non-validator need a non-validator node, or is staking enough?

**Comment by laviniat1996** [comment]
@Nexsuperne101 Those 10 nodes, that you set up as bootnodes, are the stakers. If you used the staking contracts repo to stake from them, then the rewards are stored on the address of each validator.

A bootnode is just a simple node that allows peers to discover each other. Also, not all nodes need to be bootnodes. When you generate the keys for the nodes, you can choose which ones of those to act as bootnodes.

**Comment by ghost** [comment]
thanks for your reply. So do I need to set up a non validator node for each one person that wants to stake? But instead of altering the genesis.json with the bootnode details "ip4/1.2.3.4/........../16ghfdjghdjlfgdhjk" line, they just create the .env file, add the staking-contracts details and then npm run stake and npm run register-blskey?

**Comment by laviniat1996** [comment]
@Nexsuperne101 You don't need to set up a non-validator node.

If you start with 10 validator nodes on PoA, you can still stake from them and use them as validators when you switch to PoS. 
Please note that whenever you generate keys for a node before creating the genesis file, it automatically becomes a validator. You don't need to set a node as  bootnode in order to be a validator. 

After the genesis file was created, you can add a new validator by generating the keys for a node and then:

- on PoA, by voting from the other validators for this new node to become one
- on PoS, by staking from this new node
You can find more info in our docs: [adding validator on PoA](https://wiki.polygon.technology/docs/edge/consensus/poa#adding-a-validator-to-the-validator-set) and on [PoS](https://wiki.polygon.technology/docs/edge/consensus/pos-stake-unstake#using-the-staking-smart-contract-scripts).
